Lib/profiling/sampling/gecko_collector.py

```diff
@@ -0,0 +1,451 @@
+import json
+import os
+import platform
+import sys
+import time
+from typing import Dict, List, Optional, Tuple, Any
+
+from .collector import Collector, THREAD_STATE_RUNNING
+
+
+class GeckoCollector(Collector):
+    def __init__(self, *, skip_idle=False):
+        self.skip_idle = skip_idle
+        self.start_time = time.time() * 1000  # milliseconds since epoch
+
+        # Global string table (shared across all threads)
+        self.global_strings = ["(root)"]  # Start with root
```

Comment:
ℹ️ **Performance**

Consider using a set instead of a list for `self.global_strings` if the order of strings is not important. This can improve performance for membership checks and ensure uniqueness of strings.

**Suggested fix:**
```suggestion
        self.global_strings = {"(root)"}  # Start with root
```